### PR TITLE
Add LeaderScheduleEntry

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/12?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #4** <a href="https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/4?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5** <a href="https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/5?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #9** <a href="https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/9?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #12** <a href="https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/12?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/postgres-data-handler/12?utm_source=stack-comment).

File: entries/epoch.go [hunk 1]
```diff
@@ -33,12 +32,17 @@ type PGEpochUtxoOps struct {
 
 // Convert the EpochEntry DeSo encoder to the PGEpochEntry struct used by bun.
 func EpochEntryEncoderToPGStruct(epochEntry *lib.EpochEntry, keyBytes []byte, params *lib.DeSoParams) EpochEntry {
+	var snapshotAtEpochNumber uint64
+	if epochEntry.EpochNumber >= 2 {
```
Comment (superzordon) [inline entries/epoch.go:36]: Can you add a comment explaining this?

File: entries/pkid.go [hunk 2]
```diff
@@ -95,3 +127,74 @@ func bulkDeletePkidEntry(entries []*lib.StateChangeEntry, db *bun.DB, operationT
 
 	return nil
 }
+
+func PkidBatchOperation(entries []*lib.StateChangeEntry, db *bun.DB, params *lib.DeSoParams) error {
+	// We check before we call this function that there is at least one operation type.
+	// We also ensure before this that all entries have the same operation type.
+	glog.Infof("PkidBatchOperation: Putting %d entries into the database", len(entries))
```
Comment (superzordon) [inline entries/pkid.go:134]: Lets remove this before merging, or if it's necessary for debugging in a deployed environment, just throw a FIXME above it to remove.

File: entries/pkid.go [hunk 3]
```diff
@@ -95,3 +127,74 @@ func bulkDeletePkidEntry(entries []*lib.StateChangeEntry, db *bun.DB, operationT
...
+		return errors.Wrapf(err, "entries.PostBatchOperation: Problem with operation type %v", operationType)
+	}
+	return nil
+}
+
+// bulkInsertPkid inserts a batch of PKIDs into the database.
+func bulkInsertPkid(entries []*lib.StateChangeEntry, db *bun.DB, operationType lib.StateSyncerOperationType, params *lib.DeSoParams) error {
+	// Track the unique entries we've inserted so we don't insert the same entry twice.
+	uniqueEntries := consumer.UniqueEntries(entries)
+	// NOTE: if we need to support parsing other indexes for PKIDs beyond LeaderSchedule,
```
Comment (superzordon) [inline entries/pkid.go:152]: Just for the sake of completeness, lets do a prefix filter on the uniqueEntries to just the leader schedule prefix, and use that slice in the rest of the function.

**Review by superzordon — CHANGES_REQUESTED** [review]
A few small refactors for code understandability and cleanliness, but otherwise looks good to me!

**Review by superzordon — APPROVED** [review]

**Pull request merged**